pkg/ipmi/ipmi_vm_test.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2019-2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ipmi
+
+import (
+	"testing"
+
+	"github.com/u-root/u-root/pkg/qemu"
+	"github.com/u-root/u-root/pkg/vmtest"
+)
+
+func TestIntegration(t *testing.T) {
+	o := &vmtest.Options{
+		QEMUOpts: qemu.Options{
+			Devices: []qemu.Device{
+				// This integration test requires kernel built with the following options set:
+				// CONFIG_IPMI=y
+				// CONFIG_IPMI_DEVICE_INTERFACE=y
+				// CONFIG_IPMI_WATCHDOG=y
+				// CONFIG_IPMI_SI=y
+				qemu.ArbitraryArgs{"-device", "ipmi-bmc-sim,id=bmc0"},
+				qemu.ArbitraryArgs{"-device", "pci-ipmi-kcs,bmc=bmc0"},
+			},
+		},
+	}
+	vmtest.GolangTest(t, []string{"github.com/u-root/u-root/pkg/ipmi"}, o)
+}
+
+func TestWatchdogRunningQemu(t *testing.T) {
+	i, err := Open(0)
+	if err != nil {
+		t.Fatalf("Open(0):= i,nil, not nil, %q", err)
+	}
+	defer i.Close()
+
+	if _, err := i.WatchdogRunning(); err != nil {
+		t.Errorf(`i.WatchdogRunning() = nil, not %q`, err)
+	}
+}
+func TestShutoffWatchdogQemu(t *testing.T) {
+	i, err := Open(0)
+	if err != nil {
+		t.Fatalf("Open(0):= i,nil, not nil, %q", err)
+	}
+	defer i.Close()
+
+	if err := i.ShutoffWatchdog(); err != nil {
+		t.Errorf(`i.ShutoffWatchdog() = nil, not %q`, err)
+	}
+}
+func TestGetDeviceIDQemu(t *testing.T) {
+	t.Skip("Command has unexpected End Of File")
```

Comment:
yeah what's an unexpected End Of File? that sounds like a legitimate bug or something?

Reply:
There is no information about the features of qemu bmc sim expect of that it has watchdog and shutdown feature. So prolly everything beyond that cant be tested anyway.

Reply:
```
-device ipmi-bmc-sim,id=id[,prop[=value][,...]]
Add an IPMI BMC. This is a simulation of a
hardware management interface processor that normally sits on a system. It provides a watchdog and the ability
to reset and power control the system. You need to connect this to an IPMI interface to make it useful
```

[QEMU 6.2.50 Doc, p.30/36](https://qemu.readthedocs.io/_/downloads/en/latest/pdf/)

Reply:
Started reading QEMU code to find out what I can expect from the simulator. Documentation does not give away the whole truth.